python/ray/rllib/agents/sac/models.py

```diff
@@ -0,0 +1,179 @@
+from collections import OrderedDict
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+from .squash_bijector import SquashBijector
+
+
+SCALE_DIAG_MIN_MAX = (-20, 2)
+
+
+class GaussianLatentSpacePolicy(object):
+    def __init__(self,
+                 input_shape,
+                 output_shape,
+                 hidden_layer_sizes,
+                 squash=True,
+                 activation='relu',
+                 output_activation='linear',
+                 name=None,
+                 *args,
+                 **kwargs):
+        self._squash = squash
+
+        self.condition_inputs = [tf.keras.layers.Input(shape=input_shape)]
+
+        conditions = (
+            tf.keras.layers.Concatenate(axis=-1)(self.condition_inputs)
+            if len(self.condition_inputs) > 1
+            else self.condition_inputs[0])
+
+        out = conditions
+        for units in hidden_layer_sizes:
+            out = tf.keras.layers.Dense(
+                units, *args, activation=activation, **kwargs)(out)
+
+        out = tf.keras.layers.Dense(
+            output_shape[0] * 2, *args,
+            activation=output_activation, **kwargs
+        )(out)
+
+        shift, log_scale_diag = tf.keras.layers.Lambda(
+            lambda shift_and_log_scale_diag: tf.split(
+                shift_and_log_scale_diag,
+                num_or_size_splits=2,
+                axis=-1)
+        )(out)
+
+        log_scale_diag = tf.keras.layers.Lambda(
+            lambda log_scale_diag: tf.clip_by_value(
+                log_scale_diag, *SCALE_DIAG_MIN_MAX)
+        )(log_scale_diag)
+
+        batch_size = tf.keras.layers.Lambda(
+            lambda x: tf.shape(x)[0])(conditions)
+
+        base_distribution = tfp.distributions.MultivariateNormalDiag(
+            loc=tf.zeros(output_shape),
+            scale_diag=tf.ones(output_shape))
+
+        latents = tf.keras.layers.Lambda(
+            lambda batch_size: base_distribution.sample(batch_size)
+        )(batch_size)
+
+        def raw_actions_fn(inputs):
+            shift, log_scale_diag, latents = inputs
+            bijector = tfp.bijectors.Affine(
+                shift=shift,
+                scale_diag=tf.exp(log_scale_diag))
+            actions = bijector.forward(latents)
+            return actions
+
+        raw_actions = tf.keras.layers.Lambda(
+            raw_actions_fn
+        )((shift, log_scale_diag, latents))
+
+        squash_bijector = (
+            SquashBijector()
+            if self._squash
+            else tfp.bijectors.Identity())
+
+        actions = tf.keras.layers.Lambda(
+            lambda raw_actions: squash_bijector.forward(raw_actions)
+        )(raw_actions)
```

Comment:
Could we instead use ModelCatalog.get_action_dist() to handle the action sampling so that it works with tuple action spaces as well?